src/pkg/services/m365/api/graph/service.go

```diff
@@ -392,11 +393,12 @@ func (aw *adapterWrap) Send(
 			break
 		}
 
+		err = stackWithCoreErr(ictx, err, 1)
+
 		// force an early exit on throttling issues.
-		// those retries are well handled in middleware already. We want to ensure
-		// that the error gets wrapped with the appropriate sentinel here.
-		if IsErrApplicationThrottled(err) {
-			return nil, clues.StackWC(ictx, core.ErrApplicationThrottled, err).WithTrace(1)
+		// those retries are already handled in middleware.
+		if errors.Is(err, core.ErrApplicationThrottled) {
+			return nil, err
 		}
```

Comment:
is this explicit check still required? It seems like this would fit into the default case in the switch below